### PR TITLE
fix: footer title field length expands parent component too much

### PR DESCRIPTION
**Pull request by JuanNicolasGarcia**
#### What do these changes do/fix?

This fix prevents the footer title length from expanding its parent so much where it push the Improvement and Customize Panel component off the screen in tooling.

#### How do you test/verify these changes?

This is a companion PR with (https://github.ibm.com/Watson-Discovery/discovery-tooling/pull/7397).  To test this code, checkout this PR in components and the other PR in tooling, and use `yarn link-components`.

To verify the change works, load the sample project and search `IBM`.  See the search results.  Inspect the footer title of the first result.  Manually edit the text information with the footer tag to something very long.  See that the footer title will have an `...` at the end and does not push the Improvement and Customize Panel off the screen.

Before:

8cd6a500-5cb3-11ea-96ca-30f68119ca62

After:

Screen Shot 2020-04-10 at 10 12 49 AM

### CODE REVIEW

**Comment by JuanNicolasGarcia** [comment]
Issue: https://github.ibm.com/Watson-Discovery/the-tooling-core-tracker/issues/1236

**Comment by JuanNicolasGarcia** [comment]
Will update description when other PR in tooling is ready.

File: packages/discovery-react-components/src/components/SearchResults/components/Result/Result.tsx [hunk 1]
```diff
@@ -119,7 +119,11 @@ export const Result: React.FunctionComponent<ResultProps> = ({
   const hasText = displayedText && !showTablesOnlyResults;
   const emptyResultContent = !(hasText || tableHtml);
 
-  const title = getDocumentTitle(result, resultTitleField);
+  let title = getDocumentTitle(result, resultTitleField);
+
+  if (Array.isArray(title)) {
```
Comment (H-Fin) [inline packages/discovery-react-components/src/components/SearchResults/components/Result/Result.tsx:124]: Since its not obvious, maybe a brief comment explaining why this is needed might be nice.

File: packages/discovery-react-components/src/components/SearchResults/components/Result/Result.tsx [hunk 2]
```diff
@@ -119,7 +119,11 @@ export const Result: React.FunctionComponent<ResultProps> = ({
...
+    title = title.join(', ');
```
Comment (maniax89) [inline packages/discovery-react-components/src/components/SearchResults/components/Result/Result.tsx:125]: this doesn't make sense - why would we be joining the titles with commas? shouldn't we just select the first element? commas are not i18n-friendly for other languages

Comment (JuanNicolasGarcia) [reply packages/discovery-react-components/src/components/SearchResults/components/Result/Result.tsx:125]: Honestly, I think just showing the first element is the best choice, but I was never really given any clear answer on what should be displayed when I asked about this.  Granted this is very much an edge case where someone would deliberately annotate several title fields and then search by the title field.

I can ask again to hopefully get a clear answer.

Comment (jhpedemonte) [reply packages/discovery-react-components/src/components/SearchResults/components/Result/Result.tsx:125]: I agree with the maniac: don't do this; just select the first one.

File: packages/discovery-styles/scss/components/search-results/_search-results.scss [hunk 3]
```diff
@@ -175,6 +175,7 @@ $result-table-fade-offset: calc(#{$result-element-max-height} - #{$result-table-
 }
 
 .#{$prefix}--search-result__footer__title {
+  max-width: 75%;
```
Comment (jhpedemonte) [inline packages/discovery-styles/scss/components/search-results/_search-results.scss:178]: Why is this necessary?

Comment (JuanNicolasGarcia) [reply packages/discovery-styles/scss/components/search-results/_search-results.scss:178]: If the max width of the footer title is not constrained in any way, it will actually occupy almost all of the footer space leaving no room for the collection name on the right side.

File: packages/discovery-react-components/src/components/SearchResults/components/Result/Result.tsx (hunk 2 above)
Comment (JuanNicolasGarcia) [reply packages/discovery-react-components/src/components/SearchResults/components/Result/Result.tsx:125]: This was the conclusion we reached in standup, will add this change.

File: packages/discovery-styles/scss/components/search-results/_search-results.scss (hunk 3 above)
Comment (JuanNicolasGarcia) [reply packages/discovery-styles/scss/components/search-results/_search-results.scss]: Changed to `flex: 1` and added extra margin on right of footer title.
